### DISCUSSION TITLE
[Feature] support change repository host

### DISCUSSION THREAD

**Discussion by futurist** [swiftbar/SwiftBar#342] [General]
I want to setup a private plugin repo in the company, which includes tools for internal using.

One way I can image is to change the repository host to our native location, like an IP address which not public.

That way people can just click browser/get repository in the menu (but not from github)

Is it possible?

**Comment by melonamin** [comment]
SwiftBar's repository browser stands on two API calls with the following outputs:
* [Categories](https://xbarapp.com/docs/plugins/categories.json)
* Plugins for each category, as an example category [Dev/Hosting](https://xbarapp.com/docs/plugins/Dev/Hosting/plugins.json)

You may notice that Plugins API returns the text of each plugin, SwiftBar DOESN'T use that, instead it downloads the file from GitHub.

If you can replicate the API and make your plugins available through http\https then it would be relatively simple for me to add support for custom repositories.

We can take an alternative route and standardize a separate API for private repos, which let say, could be just a git repo with a special index.json file, but this needs more time and thought

**Reply by futurist** [reply]
I see, can you make current hardcoded xbarapp/github repository URL be configurable? That way we can try to implement a version.

**Reply by melonamin** [reply]
I created an issue to address this #351, let's move the conversation there. I'll ping you when I have a beta
